<?xml version="1.0" encoding="utf-8"?>

<launch>
  <arg name="baudrate" default="2000000" />
  <arg name="onlydistdata" default="false" />
  <arg name="use_logger" default="false" />

  # laser1
  <arg name="serial_port1" default="/dev/laser_T" />
  <arg name="laser_name1" default="T" />
  <arg name="laser_link1" default="laser_link_" />
  <arg name="lensfocus1" default="8.0" />

  # laser2
  <arg name="serial_port2" default="/dev/laser_K" />
  <arg name="laser_name2" default="K" />
  <arg name="laser_link2" default="laser_link_" />
  <arg name="lensfocus2" default="8.0" />

  <node name="jsk_laser_node_1" pkg="jsk_laser" type="jsk_laser_node" output="screen">
    <rosparam subst_value="true">
      serial_port: $(arg serial_port1)
      baudrate: $(arg baudrate)
      laser_name: $(arg laser_name1)
      onlydistdata: $(arg onlydistdata)
      lensfocus: $(arg lensfocus1)
      laser_link: $(arg laser_link1)
    </rosparam>

  </node>

  <node name="jsk_laser_node_2" pkg="jsk_laser" type="jsk_laser_node" output="screen">
    <rosparam subst_value="true">
      serial_port: $(arg serial_port2)
      baudrate: $(arg baudrate)
      laser_name: $(arg laser_name2)
      onlydistdata: $(arg onlydistdata)
      lensfocus: $(arg lensfocus2)
      laser_link: $(arg laser_link2)
    </rosparam>
  </node>

  <group if ="$(arg use_logger)" >
    <node name="laser_logger_1" pkg="jsk_laser" type="logger.py" output="screen" >
      <param name="laser_name" value="$(arg laser_name1)" />
      <param name="laser_link" value="$(arg laser_link1)" />
    </node>

    <node name="laser_logger_2" pkg="jsk_laser" type="logger.py" output="screen" >
      <param name="laser_name" value="$(arg laser_name2)" />
      <param name="laser_link" value="$(arg laser_link2)" />
    </node>
  </group>

  <node  pkg="tf" type="static_transform_publisher" name="scan1_broadcaster" args="0.22 0 -0.05 1.5208 0 0  /link2 /laser_link_T  20" />
  <node  pkg="tf" type="static_transform_publisher" name="scan2_broadcaster" args="0.218 0 -0.05 -1.5608 3.14159 0  /link4 /laser_link_K  20" />

  <node pkg = "aerial_recognition" name = "laser_inertial_parameters" type = "laser_inertial_parameters.py" >
    <param name = "laser1_parent_link" value = "1" />
    <param name = "laser1_mass" value = "0.06" />
    <param name = "laser1_offset" value = "-0.09" />
    <param name = "laser2_parent_link" value = "3" />
    <param name = "laser2_mass" value = "0.055" />
    <param name = "laser2_offset" value = "-0.08" />

  </node>

</launch>
